src/lib.rs

```diff
@@ -73,11 +79,28 @@ pub fn is_terminal<T: IsTerminal>(this: T) -> bool {
 impl<Stream: AsFd> IsTerminal for Stream {
     #[inline]
     fn is_terminal(&self) -> bool {
-        #[cfg(any(unix, target_os = "wasi"))]
+        #[cfg(all(any(unix, target_os = "wasi"), feature = "rustix"))]
         {
             rustix::termios::isatty(self)
         }
 
+        #[cfg(all(
+            any(unix, target_os = "wasi"),
+            feature = "libc",
+            not(feature = "rustix")
+        ))]
+        unsafe {
+            libc::isatty(self.as_fd().as_raw_fd()) != 0
+        }
+
+        #[cfg(all(
+            any(unix, target_os = "wasi"),
+            not(any(feature = "rustix", feature = "libc"))
+        ))]
+        {
+            compile_error!("either \"rustix\" or \"libc\" must be enabled");
```

Comment:
does `libc` really need to be optional? `rustix` has `libc` as a dependency anyway, so by opting for the `rustix` backend you don't get rid of `libc` from the dependency tree anyway. If you make `libc` non-optional and it uses it by default unless a `rustix` feature is enabled, there won't be any combination of features yielding a compiler error.